server/controllers/excelController.js

```diff
@@ -205,184 +205,172 @@ excelController.getRelationships = async (req, res, next) => {
 
     rows.push(tempObj)
   }
+  res.locals.inputRows = rows;
+  // REPLACE WITH BRIAN'S INPUT AND REMOVE THIS ^^^^
 
-  const tables = {};
-  for (let obj of output) {
-    // add id col to each table
-    obj.columns.unshift({
-      _id: {
-        primaryKey: true, type: 'INT'
-      }
-    })
 
-    // un-nest object for the sake of later table lookup
-    const tempObj = {};
-    for (let colObj of obj.columns) {
-      for (let col in colObj) {
-        tempObj[col] = colObj[col];
-      }
+  try {
+    for (let obj of res.locals.output) {
+      // add id col to each table
+      obj.columns.unshift({
+        _id: {
+          primaryKey: true, type: 'INT'
+        }
+      })
```

Comment:
Sorry if I was being unclear earlier - if we add a primary key to each table than the `type` should be `SERIAL`. However, if the column is a _foreign key_, then the `type` should be `INT` (which you are doing correctly in L314 below).  You can look [here](https://github.com/ExcQL/ExcQL/blob/dd38014b5b7a9d6b4e30569dd1aac86b1501854f/__tests__/sqlController.js) to see what I mean.